### ISSUE TITLE
🐛 BUG: FetchContent seems to not support fetching a single file anymore

### ISSUE THREAD

**Issue by TimWeidner** [snowpackjs/astro#1012]
### What package manager are you using?

npm

### What operating system are you using?

Windows

### Describe the Bug

I am not sure if this is a regression bug, but `Astro.fetchContent('../pages/posts/test.md');` throws the following error: `Error: Parse error @:1:1`
Astro.fetchContent() was able to fetch a single `.md` file in earler versions of astro (I am updating from ~0.15 to 0.18.8). The docs now state you can give it a glob url and it'll return an array of entries, but I am neither sure if the original usecase was dropped on purpose, nor what I am supposed to use instead (Except `Array.find()` which you'll agree is anything but optimal)

### Steps to Reproduce

1. Try to const stuff = Astro.fetchContent(path-to-file) on a single file
2. Open the page in a browser
5. Error! Returns `Error: Parse error @:1:1` (expected to be able to use the stuff variable)

<details>
  <summary>Full Error</summary>
  
```
 [22:18:41] [snowpack] Build Result Error: There was a problem with a file build result.
[executing astro] Error: Parse error @:1:1
    at parse (D:\dev\Website\node_modules\snowpack\node_modules\es-module-lexer\dist\lexer.cjs:1:402)
    at scanCodeImportsExports (D:\dev\Website\node_modules\snowpack\lib\cjs\rewrite-imports.js:21:29)
    at transformEsmImports (D:\dev\Website\node_modules\snowpack\lib\cjs\rewrite-imports.js:37:27)
    at Object.transformFileImports (D:\dev\Website\node_modules\snowpack\lib\cjs\rewrite-imports.js:108:16)
    at FileBuilder.resolveImports (D:\dev\Website\node_modules\snowpack\lib\cjs\build\file-builder.js:116:48)
    at async Object.loadUrl (D:\dev\Website\node_modules\snowpack\lib\cjs\commands\dev.js:640:39)
    at async load (D:\dev\Website\node_modules\snowpack\lib\cjs\commands\dev.js:217:28)
    at async D:\dev\Website\node_modules\snowpack\lib\cjs\ssr-loader\index.js:60:28
    at async D:\dev\Website\node_modules\snowpack\lib\cjs\ssr-loader\index.js:83:23
    at async initializeModule (D:\dev\Website\node_modules\snowpack\lib\cjs\ssr-loader\index.js:88:39)
    at async Object.importModule (D:\dev\Website\node_modules\snowpack\lib\cjs\ssr-loader\index.js:166:25)
    at async load (file:///D:/dev/Website/node_modules/astro/dist/runtime.js:76:17)
    at async Server.<anonymous> (file:///D:/dev/Website/node_modules/astro/dist/dev.js:21:20) {
  idx: 0
}
```
</details>

**Comment by sarah11918** [comment]
Just confirming that I can reproduce. If I add ```let singlePost = Astro.fetchContent('../pages/posts/fourth-post.md');``` (without even trying to do anything with the result) in my code fence of an .astro file that is otherwise successfully fetching ALL .md posts just fine, I get the same parse error.

**Comment by FredKSchott** [comment]
I can confirm that this is broken by v0.18.0, where we moved this implementation from a custom globbing system to just use `import.meta.globEager` internally. This is a Snowpack feature that was never really documented, but matches Vite's `import.meta.globEager` support: https://vitejs.dev/guide/features.html#glob-import

This should be fixed in Snowpack!

unrelated: the parse error @:1:1 feels like a different bug, fo some reason the file is failing to build and something inside of Astro is returning an empty string instead. But I'm not sure where that could be happening in the latest version of Astro.

**Comment by jasikpark** [comment]
what's the status of this bug?

**Comment by tony-sull** [comment]
I'm not actually able to reproduce this one, maybe it was fixed in a more recent release?  I created a fresh Astro project with the basic starter template and was able to use both `import.meta.globEager('../data.json')` and `import.meta.globEager('.pages/test.md')` to pull a couple test files into the index page.

@TimWeidner @sarah11918, are either of you still able to reproduce the bug?

**Comment by tony-sull** [comment]
Just got this one to repro locally in Astro with `Astro.fetchContent`.  Interestingly I can't actually get it to repro with `import.meta.globEager` directly, or in a Snowpack app directly

I'll see if I can track this one down, seems to be very specific to the `Astro.fetchContent` logic itself

**Comment by tony-sull** [comment]
I may have actually spoken too soon there, I think `npm init astro` used an older release of Astro when I managed to reproduce it.

@TimWeidner Can you see if this still reproduces for you on the latest release of Astro?

**Comment by FredKSchott** [comment]
Added a compiler-v2 label to make sure that we confirm in v2, unless @TimWeidner  can confirm sooner that this is fixed

**Comment by FredKSchott** [comment]
Thanks for the report! We believe that this has been fixed in astro v0.21, going out today. Let us know if you can still reproduce the issue after updating to v0.21 and we'll be happy to take a look.
